src/runtime-rs/crates/resource/src/share_fs/share_uds.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2024 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//#![deny(warnings)]
+use agent::sock::{self, Listener};
+use anyhow::{anyhow, Result};
+use slog::info;
+use std::{collections::HashMap, sync::Arc};
+use tokio;
+use tokio::net::UnixStream;
+use tokio::sync::Notify;
+
+pub type SockPort = u32;
+pub const SOCK_BASE_PORT: SockPort = 1080;
+
+pub const MAX_RETRY: u32 = 32;
```

Comment:
why do we retry so many times?
is it supposed to fail a lot?

Reply:
Hi @liubogithub   The retry here is to take care of the situation where qemu uses vhost vsock. Because vhost-vsock does not have network ns isolation, different pods on the same node may use the same port, so we have to keep retrying to find an unused port. Of course, this usage is not very elegant, but for now, in order to make qemu work, we can only use it this way temporarily. When vhost-vsock supports network ns isolation in the future, we can remove this part of the retry code.